extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/search/YoutubeSearchPagingTest.java

```diff
@@ -0,0 +1,78 @@
+package org.schabi.newpipe.extractor.services.youtube.search;
+
+import static java.util.Collections.singletonList;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.schabi.newpipe.extractor.ServiceList.YouTube;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.schabi.newpipe.DownloaderTestImpl;
+import org.schabi.newpipe.extractor.InfoItem;
+import org.schabi.newpipe.extractor.ListExtractor;
+import org.schabi.newpipe.extractor.NewPipe;
+import org.schabi.newpipe.extractor.services.youtube.extractors.YoutubeSearchExtractor;
+import org.schabi.newpipe.extractor.services.youtube.linkHandler.YoutubeSearchQueryHandlerFactory;
+
+public class YoutubeSearchPagingTest {
+    private static ListExtractor.InfoItemsPage<InfoItem> page1;
+    private static ListExtractor.InfoItemsPage<InfoItem> page2;
+    private static Set<String> urlList1;
+    private static Set<String> urlList2;
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        NewPipe.init(DownloaderTestImpl.getInstance());
+
+        YoutubeSearchExtractor extractor = (YoutubeSearchExtractor) YouTube.getSearchExtractor("cirque du soleil",
+                singletonList(YoutubeSearchQueryHandlerFactory.VIDEOS), null);
+
+        extractor.fetchPage();
+        page1 = extractor.getInitialPage();
+        urlList1 = extractUrls(page1.getItems());
+        assertEquals("page with items loaded", 20, page1.getItems().size());
```

Comment:
Can we always assume, that the item count is 20? I'd suggest to choose something like > 15 to not fail when YouTube changes algos again.

Reply:
I don't know, if this 20 will be 20 forever, so I could rewrite the tests to check for page sizes are equals, and the unique videos are returned.